lib/miq_expression/tag.rb

```diff
@@ -1,16 +1,24 @@
-class MiqExpression::Tag
-  def self.parse(tag)
-    klass, ns = tag.split(".")
-    klass, ns = "nil", klass if ns.nil? # support managed-label
-    ns = "/" + ns.split("-").join("/")
-    ns = ns.sub(/(\/user_tag\/)/, "/user/") # replace with correct namespace for user tags
-    new(klass.safe_constantize, ns)
+class MiqExpression::Tag < MiqExpression::Field
+  TAG_REGEX = /
+(?<model_name>([[:alnum:]]*(::)?)+)
+\.(?<associations>([a-z_]+\.)*)
+(?<namespace>\bmanaged|user_tag\b)
+-(?<column>[a-z]+[_[:alnum:]]+)
```

Comment:
there are these differences `Field`:
- model_name can start with [lower charater](https://github.com/ManageIQ/manageiq/pull/13769#discussion_r100331381).
- determination of namespace `managed` and `user_tag`